src/memory_region.rs

```diff
@@ -32,22 +32,14 @@ use std::{
     Host:  frame 0 | frame 1 | frame 2 | ...
 */
 
-/// The state of a memory region.
-#[derive(Debug, Copy, Clone, Default, Eq, PartialEq)]
-pub enum MemoryState {
-    /// The memory region is readable
-    #[default]
-    Readable,
-    /// The memory region is writable
-    Writable,
-    /// The memory region is writable but must be copied before writing. The
-    /// carried data can be used to uniquely identify the region.
-    Cow(u64),
+/// Callback executed before generate_access_violation()
+pub type AccessViolationHandler = Box<dyn Fn(u32) -> Result<u64, ()>>;
+/// Fail always
+#[allow(clippy::result_unit_err)]
+pub fn default_access_violation_handler(_: u32) -> Result<u64, ()> {
+    Err(())
```

Comment:
The usage of the function is to try to write to the memory and copy if it is readonly, right? I believe the old name `MemoryCowCallback` (or something related) is clearer on the intentions of this function.

Reply:
I can split the renaming into a separate PR. But in the future it is not only for CoW (readonly to writable) but also to map in unmapped address space (for growing the realloc padding).

Reply:
Yes, another PR is good.

> but also to map in unmapped address space

`MemoryCowCallback` isn't the best option considering that, but I still prefer it to describe what the function does, instead of the error it might return when it can't do what it is supposed to.

Reply:
Well the naming scheme is similar to the signal segmentation fault (SIGSEGV) [signal handler](https://en.wikipedia.org/wiki/C_signal_handling). That one is also named after the signal / error message.